### PR TITLE
Passwd build rule prototype

### PR DESCRIPTION
**Pull request by dlorenc**
Here's a quick version of what we discussed in #14

### CODE REVIEW

File: tools/BUILD [hunk 1]
```diff
@@ -0,0 +1,6 @@
+load(":passwd.bzl", "passwd_file")
+
+passwd_file(
```
Comment (dlorenc) [inline tools/BUILD]: I'll remove this, it was just for testing.

File: tools/passwd.bzl [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  )
+  build_tar = ctx.executable.build_tar
+  args = [
+      "--output=" + ctx.outputs.tar.path,
+      "--file=%s=/etc/passwd" % ctx.outputs.out.path
+  ]
+  arg_file = ctx.new_file(ctx.attr.name + ".args")
+  ctx.file_action(arg_file, "\n".join(args))
+
+  ctx.action(
```
Comment (mattmoor) [inline tools/passwd.bzl]: @damienmg What's the best way to send this through `pkg_tar`

File: tools/passwd.bzl [hunk 3]
```diff
@@ -0,0 +1,56 @@
+load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar")
```
Comment (mattmoor) [inline tools/passwd.bzl]: tl;dr I think we should add this to `bazelbuild/rules_docker`.

Perhaps under `docker/contrib` for now.  It should be very straightforward to add a rule that uses it under `docker/testdata` and verifies the result under `docker/build_test.sh`.

WDYT?

Comment (dlorenc) [inline tools/passwd.bzl]: SGTM. I'll move it over there with a test, then add it to our base here.

**Comment by dlorenc** [comment]
Closing, this was merged upstream.

**Pull request closed without merging**